cmd/bcpd/app/codec.proto

```diff
@@ -13,7 +14,20 @@ import "github.com/iov-one/weave/x/validators/codec.proto";
 // Tx contains the message
 message Tx {
   // msg is a sum type over all allowed messages on this chain.
-  oneof sum{
+  Sum sum = 1;
+  // fee info, autogenerates GetFees()
+  cash.FeeInfo fees = 2;
+  // signatures, autogenerates GetSignatures()
+  repeated sigs.StdSignature signatures = 3;
+  // preimage for hashlock, autogenerates GetPreimage
+  bytes preimage = 4;
+  // id of multisig contract, autogenerates GetMultisig
+  repeated bytes multisig = 5;
+}
+
+
+message Sum {
```

Comment:
Is `Sum` a union type (`Sum` queals `cash.SendMsg` or `namecoin.NewTokenMsg` or ... or `BatchMsg`)? Then I think `Union` would be the better name

Reply:
It is